### DISCUSSION TITLE
Proposal : Add more details of API to docs

### DISCUSSION THREAD

**Discussion by coyjiv** [keycloakify/keycloakify#95] [Ideas]
### The docs page lacks information for creating your own theme.

- There are no details on which props you should use for different pages, or _how to add the theme_ to keycloak. 
- It is really hard to figure out how it works from the examples that are shown in docs, considering that in one demo example there aren't any forms. 

Also, there is a typo in "Developpement" https://docs.keycloakify.dev/developpement

**Comment by garronej** [comment]
Hi @coyjiv 

> There are no details on which props you should use for different pages.  

Keycloakify respects the convention established by Keycloak, it just provides a way to do it with react.  

> or how to add the theme to keycloak.  

Indeed this isn't in the doc but when you run `yarn keycloak` is printed on the console detailed information about how to proceed to import the theme in Keycloak.  

> It is really hard to figure out how it works from the examples that are shown in docs, considering that in one demo example there aren't any forms.  

The real documentation are the demo projects:  

https://github.com/garronej/keycloakify-demo-app for CSS level customization 

and 

https://github.com/garronej/keycloakify-demo-app/tree/look_and_feel for Component level customization.  

You also have [the source code](https://github.com/InseeFrLab/onyxia-web) of [this production app](https://datalab.sspcloud.fr/home) that you can consult.   

Lastly you can download the source files of the default themes with:  

```bash
npx -p keycloakify download-builtin-keycloak-theme
```

Thanks for the feedback but be aware that this is just me, and I am doing this on my free time.   

Best regards

**Reply by coyjiv** [reply]
Thank you for your response, but I guess it would be much easier for new users of keycloakify to find all the information that they needed directly from the docs page, is there is a way to make a pull request on docs website, so I could specify all the things that you've mentioned?

**Reply by garronej** [reply]
You are very welcome to submit a PR to the documentation.  
It's [on the `docs` branch of this repo](https://github.com/InseeFrLab/keycloakify/tree/docs).  
I have already made some changes based on your feedback but you are welcome to add anything you'd see fit.
